### DISCUSSION TITLE
How to tell PMM to write logs to /tmp instead of logs dir / how to point logs dir to /tmp ?

### DISCUSSION THREAD

**Discussion by gitttt54** [meisnate12/Plex-Meta-Manager#1350] [Q&A]
Hello,

I want the PMM logs not to be written to my SSD (where the `/config` from the docker configuration points to), but to `/tmp` (ramdisk).

The PMM docker container points `/config` to `/mnt/users/appdata/pmm`
so I created a symlink `logs` from `/mnt/users/appdata/pmm` to `/tmp/pmm.logs`
via this script
```
mkdir -p /tmp/pmm.logs
[ -d "/mnt/user/appdata/pmm/logs" ] && rm -rf /mnt/user/appdata/pmm/logs
[ ! -d "/mnt/user/appdata/pmm/logs" ] && ln -s /tmp/pmm.logs /mnt/user/appdata/pmm/logs
```

Now, when starting pmm, I get this error:
```
root@UnRaid:~# docker exec -it pmm python plex_meta_manager.py --run -c config/config.yml --no-missing -is
Traceback (most recent call last):
  File "//plex_meta_manager.py", line 129, in <module>
    logger = MyLogger("Plex Meta Manager", default_dir, screen_width, divider[0], ignore_ghost, test or debug, trace)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/modules/logs.py", line 53, in __init__
    os.makedirs(self.log_dir, exist_ok=True)
  File "<frozen os>", line 225, in makedirs
FileExistsError: [Errno 17] File exists: '/config/logs'
```

It seems, PMM doesn't like a symlink where it expects a folder?

So I tried another approach:
I also tried to set the folder from the docker config, like `-v '/tmp/pmm.logs/':'/config/logs':'rw' `, but that also did not work. PMM ignores my wish and still writes to `/mnt/users/appdata/pmm/logs` instead of  `/tmp/pmm.logs`.

```
docker run
  -d
  --name='pmm'
  --net='br0'
  -e TZ="Europe/Berlin"
  -e HOST_OS="Unraid"
  -e HOST_HOSTNAME="UnRaid"
  -e HOST_CONTAINERNAME="pmm"
  -l net.unraid.docker.managed=dockerman
  -l net.unraid.docker.icon='https://raw.githubusercontent.com/meisnate12/Plex-Meta-Manager/master/docs/_static/favicon.png'
  -v '/mnt/user/appdata/pmm/':'/config':'rw'
  -v '/tmp/pmm.logs/':'/config/logs':'rw' 'meisnate12/plex-meta-manager'
The command finished successfully!
```

Maybe I did something wrong here?

What might be the easiest way to have PMM write its logs to the hosts `/tmp` folder?

Thank you!

**Comment by mikenobbs** [comment]
Not sure if the location makes a difference but I re-map all my logs without issue. My compose looks like this

```
volumes:
      - ${CONFDIR}/plex-meta-manager:/config:rw
      - ${LOGDIR}/plex-meta-manager:/config/logs
```

In theory `${LOGDIR}` could point to any location and will work providing the permissions are correct.

**Comment by gitttt54** [comment]
Thank you,
 your example shows, that you map your logs to the default dir, so I guess, thats why its working.

Would you mind to test it with a path _outside_ of the docker container?

I chmodded everything, including the symlink to 777, so permissions are good.

I just don't know, if PMM allows to point the logs to somewhere else or what exactly PMM complains about in my first post.

`File exists: '/config/logs'` _seems_ to me as if PMM tries to create the logs folder, but can not, because a symlink is already in that place and PMM can not handle that. 

I only found this wiki page for the [unrad walkthrough](https://metamanager.wiki/en/latest/home/guides/unraid.html) and it seems to me, that the path is fixed?

> Once finished, run the container. Voila! Logs are located in `yourhostpath/logs`.

**Reply by mikenobbs** [reply]
Can confirm it still works. Changed my path to `/media/storage/logs/plex-meta-manager:/config/logs` which is the mount where I store all my media, completely separate location to my docker location which is `~/docker/configs`, and as soon as I ran the container the log appeared in that location

**Reply by gitttt54** [reply]
Thank you very much, 
would you mind to spend a little more time and create a symlink named `logs` in the config (I guess its ~/docker/configs) folder, pointing to /media/storage/logs/plex-meta-manager/logs
like (If I got your paths right):
`ln -s /media/storage/logs/plex-meta-manager/logs ~/docker/configs/logs`
and then start PMM to see what happens?

I still do not understand why I got the above PMM error. (and why I can't set up my docker container just as yours, but thats most probably my incompetence...)

**Comment by gitttt54** [comment]
I recognized, that after a restart of the docker container, everything worked as expected. %-/

Still, PMM creates a folder "logs", but it stays empty and PMM writes everything to the symlinked folder now.
That said, I can't use a symlink named "logs" (throws the above error), but nothingness, I can use "logs2" or something. So no problem.

_Note to myself:
If you ever change/edit a symlink, you have to restart the docker container to reflect the changes of the symlink inside the container.
The container seems not to care about the changed content of the symlink, even if the symlink existed in the first place.
There must be some pointers the container remembers and not the actual symlink.... computer junk_ 

Thank you @mikenobbs for helping!
